### PR TITLE
move transform gizmo to spatial package, use spatial input for transform gizmo

### LINKED ISSUE #9775: Transform Gizmo uses Spatial Input

### LINKED ISSUE #9783: Transform Gizmo has it's own spatial input heuristic

### LINKED ISSUE #9819: [Bug] Releasing gizmo control selects object behind it

### PR DESCRIPTION
**Pull request by SYBIOTE**
This pull request fixes the issue with the Transform Gizmo using Spatial Input. The changes include adding the TransformGizmoTagComponent to the relevant entities and disabling the TransformGizmo layer for raycasting when necessary.

Fixes #9775
Fixes #9819
Fixes [#9783](https://github.com/EtherealEngine/etherealengine/issues/9783)

### CODE REVIEW

**Comment by DanielBelmes** [comment]
Does this fix the UX issues I reported on the other PR. Also in the future you can just do changes on the original PR so we have one source of truth. I don't think we need a new PR for a couple of fixes.

File: packages/spatial/src/input/systems/ClientInputSystem.tsx [hunk 1]
```diff
@@ -45,6 +45,7 @@ import { defineSystem } from '@etherealengine/ecs/src/SystemFunctions'
 import { InputSystemGroup } from '@etherealengine/ecs/src/SystemGroups'
 import { EngineState } from '@etherealengine/spatial/src/EngineState'
 import { EntityTreeComponent } from '@etherealengine/spatial/src/transform/components/EntityTree'
+import { TransformGizmoTagComponent } from '../../../../editor/src/classes/TransformGizmoControlledComponent'
```
Comment (HexaField) [inline packages/spatial/src/input/systems/ClientInputSystem.tsx]: this is breaking our import hierarchy

File: packages/editor/src/functions/gizmoHelper.ts [hunk 2]
```diff
@@ -884,77 +893,40 @@ function pointerUp(pointer, gizmoEntity) {
...
-  })
-
-  pointerHover(getPointer(event), gizmoEntity)
-  pointerDown(getPointer(event), gizmoEntity)
+  pointerHover(gizmoEntity)
+  pointerDown(gizmoEntity)
 }
 
-export function onPointerMove(event, gizmoEntity) {
+export function onPointerMove(gizmoEntity) {
```
Comment (HexaField) [inline packages/editor/src/functions/gizmoHelper.ts:914]: please make sure all arguments have correct types

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/spatial/src/input/systems/ClientInputSystem.tsx (hunk 1 above)
Comment (SYBIOTE) [reply packages/spatial/src/input/systems/ClientInputSystem.tsx]: hmm, i suppose ill move the gizmo to the sptial package within this PR

**Review by HexaField — APPROVED** [review]

**Pull request merged**